v4-client-js/src/clients/composite-client.ts

```diff
@@ -1233,25 +1237,93 @@ export class CompositeClient {
     gasAdjustment?: number,
     memo?: string,
   ): Promise<BroadcastTxAsyncResponse | BroadcastTxSyncResponse | IndexedTx> {
-    return this.validatorClient.post.createMarketPermissionless(ticker, subaccount, broadcastMode, gasAdjustment, memo);
+    return this.validatorClient.post.createMarketPermissionless(
+      ticker,
+      subaccount,
+      broadcastMode,
+      gasAdjustment,
+      memo,
+    );
   }
 
   async addAuthenticator(
     subaccount: SubaccountInfo,
     authenticatorType: AuthenticatorType,
     data: Uint8Array,
   ): Promise<BroadcastTxAsyncResponse | BroadcastTxSyncResponse | IndexedTx> {
-    return this.validatorClient.post.addAuthenticator(subaccount, authenticatorType, data)
+    // Validate the provided authenticators before sending to the validator
+    const authenticator: Authenticator = {
+      type: authenticatorType,
+      config: JSON.parse(new TextDecoder().decode(data)),
+    };
+    if (!this.validateAuthenticator(authenticator)) {
+      throw new Error(
+        'Invalid authenticators: Nested authenticators must include a SIGNATURE_VERIFICATION authenticator.',
+      );
+    }
+
+    return this.validatorClient.post.addAuthenticator(subaccount, authenticatorType, data);
   }
 
   async removeAuthenticator(
     subaccount: SubaccountInfo,
     id: Long,
   ): Promise<BroadcastTxAsyncResponse | BroadcastTxSyncResponse | IndexedTx> {
-    return this.validatorClient.post.removeAuthenticator(subaccount, id)
+    return this.validatorClient.post.removeAuthenticator(subaccount, id);
   }
 
-  async getAuthenticators(address: string): Promise<GetAuthenticatorsResponse>{
+  async getAuthenticators(address: string): Promise<GetAuthenticatorsResponse> {
     return this.validatorClient.get.getAuthenticators(address);
   }
+
+  // Function to validate that all authentication paths require SIGNATURE_VERIFICATION
+  private validateAuthenticator(
```

Comment:
Could we add some unit test for this? Could cover common success/failure examples.